assets/javascripts/modules/addRemove.js

```diff
@@ -40,15 +40,26 @@ var DEFAULT_ADD_BTN_TEXT = 'Add',
     addRemoveInput = '[data-add-remove-input]',
     template = '[data-template]',
     addButton = '[data-add-btn]',
-    maxItems = 'data-max';
+    maxItems = 'data-max',
+    cachedTemplate = [];
 
 var init = function () {
   var $addRemoveContainers = $(addRemoveContainer);
 
   if ($addRemoveContainers.length) {
 
     $addRemoveContainers.each(function (i, container) {
-      var $container = $(container);
+      var $container = $(container),
+          $cloneableItem = $container.find('[' + $container.attr('data-template-item') + ']');
+
+      $container.data('index', i);
```

Comment:
We can't rely the index of it's position in the `$addRemoveContainers` array rather than using `data`?